ocs_ci/deployment/provider_client/storage_client_deployment.py

```diff
@@ -0,0 +1,630 @@
+        self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.UX_BACKEND_SERVER_LABEL,
+            resource_count=1,
+            timeout=180,
+        )
+        # Native storageclients are created as part of ODF 4.16 subscription and each of rbd and
+        # cephfs storageclaims gets created automatically with the storageclient creation
+        if self.ocs_version >= version.VERSION_4_16:
+            # Validate native client is created in openshift-storage namespace
+            self.deployment.wait_for_csv(
+                self.ocs_client_operator, constants.OPENSHIFT_STORAGE_NAMESPACE
+            )
+            log.info(
+                f"Sleeping for 30 seconds after {self.ocs_client_operator} created"
+            )
+
+            # Validate storageclaims are Ready and associated storageclasses are created
+            verify_storageclient()
+
+            # Validate cephblockpool created
+            assert verify_block_pool_exists(
+                constants.DEFAULT_BLOCKPOOL
+            ), f"{constants.DEFAULT_BLOCKPOOL} is not created"
+            assert (
+                self.rados_utils.verify_cephblockpool_status()
+            ), "the cephblockpool is not in Ready phase"
+
+            # Validate radosnamespace created and in 'Ready' status
+            assert (
+                self.rados_utils.check_phase_of_rados_namespace()
+            ), "The radosnamespace is not in Ready phase"
+
+            # Validate storageclassrequests created
+            storage_class_classes = get_all_storageclass_names()
+            for storage_class in self.storage_class_claims:
+                assert (
+                    storage_class in storage_class_classes
+                ), "Storage classes ae not created as expected"
+
+        else:
+            # Create ODF subscription for storage-client
+            self.odf_installation_on_client()
+
+            # Fetch storage provider endpoint details
+            storage_provider_endpoint = self.ocp_obj.exec_oc_cmd(
+                (
+                    f"get storageclusters.ocs.openshift.io -n {config.ENV_DATA['cluster_namespace']}"
+                    + " -o jsonpath={'.items[*].status.storageProviderEndpoint'}"
+                ),
+                out_yaml_format=False,
+            )
+            log.info(f"storage provider endpoint is: {storage_provider_endpoint}")
+
+            # Create Network Policy
+            self.create_network_policy(
+                namespace_to_create_storage_client=constants.OPENSHIFT_STORAGE_CLIENT_NAMESPACE
+            )
+            onboarding_token = self.onboarding_token_generation_from_ui()
+
+            # Create native storage client
+            create_storage_client(
+                storage_provider_endpoint=storage_provider_endpoint,
+                onboarding_token=onboarding_token,
+            )
+
+        # Check nooba db pod is up and running
```

Comment:
It is a good practice to separate installation and complete verification

Reply:
Have separated verification method.